daemon/images.go

```diff
@@ -271,7 +271,9 @@ func (daemon *Daemon) SquashImage(id, parent string) (string, error) {
 		rootFS := image.NewRootFS()
 		parentImg = &image.Image{RootFS: rootFS}
 	}
-
+	if !system.IsOSSupported(img.OperatingSystem()) {
+		return "", errors.Wrap(err, system.ErrNotSupportedOperatingSystem.Error())
+	}
```

Comment:
How does this relate to the rest of the PR?

Reply:
It's a safety check to ensure that we don't panic on L277 where the layerstores array is indexed by the operating system. Strictly it's not related to this PR, it should have been in https://github.com/moby/moby/pull/34859, for which this PR is a direct follow up. Happy to move it out to a separate PR if you prefer.